src/todo_file/src/state.rs

```diff
@@ -0,0 +1,82 @@
+use std::{
+	fmt::{Display, Formatter},
+	path::{Path, PathBuf},
+};
+
+use crate::errors::{FileReadErrorCause, IoError};
+
+/// Describes the state of rebase when editing the rebase todo file.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[allow(clippy::exhaustive_enums)]
+pub enum State {
+	/// Editing todo at start of a rebase.
+	Initial,
+	/// Editing todo in the middle of a rebase with --edit.
+	Edit,
+	/// Editing the todo file for git-revise
+	Revise,
+}
+
+pub(crate) fn detect_state(filepath: &Path) -> Result<State, IoError> {
+	if filepath.ends_with("git-revise-todo") {
+		return Ok(State::Revise);
+	}
+	if let Some(parent) = filepath.parent() {
+		if parent.join("stopped-sha").try_exists().map_err(|err| {
+			IoError::FileRead {
+				file: PathBuf::from(parent),
+				cause: FileReadErrorCause::from(err),
+			}
+		})? {
+			return Ok(State::Edit);
+		}
+	}
+	return Ok(State::Initial);
+}
+
+impl State {}
+
+impl Display for State {
+	#[inline]
+	fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+		write!(f, "{}", match *self {
+			Self::Initial => "initial",
+			Self::Edit => "edit",
+			Self::Revise => "revise",
+		})
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use rstest::rstest;
+
+	use super::*;
+	use crate::testutil::{with_todo_revise_file, TodoFileTestDirContext, with_todo_rebase_edit_file, with_todo_rebase_edit_file_stopped};
```

Comment:
There are a few formatting issues that can be auto fixed, this line is a good example.

It seems like you are using VSCode, and there is an [official plugin](https://code.visualstudio.com/docs/languages/rust) for it, that can run format on save.


The other option, is to run  `cargo make format-apply`, and it will fix any formatting issues.